internal/events/logprocessor.go

```diff
@@ -96,9 +98,8 @@ func (lp *logProcessor) getBlockHWM() big.Int {
 func (lp *logProcessor) markNoEvents(blockNumber *big.Int) {
 	lp.hwnSync.Lock()
 	if lp.highestDispatched.Cmp(&lp.blockHWM) < 0 {
-		// Nothing in-flight, its safe to update the HWM
-		lp.blockHWM.Set(blockNumber)
-		log.Debugf("%s: HWM: %s", lp.subID, lp.blockHWM.String())
+		// Nothing in-flight, its safe to update the HWM - to one after the block we're up to
+		lp.blockHWM.Set(new(big.Int).Add(blockNumber, big.NewInt(1)))
```

Comment:
Without this, each restart we were going back one block.

Reply:
Wow, definitely missed this the last time I worked on fixing the restart!